lib/src/web_socket_handler.dart

```diff
@@ -78,26 +78,26 @@ class WebSocketHandler {
       if (protocol != null) sink.add('Sec-WebSocket-Protocol: $protocol\r\n');
       sink.add('\r\n');
 
-      _onConnection(
+      _onConnection?.call(
           WebSocketChannel(channel, pingInterval: _pingInterval), protocol);
     });
 
     // [request.hijack] is guaranteed to throw a [HijackException], so we'll
     // never get here.
     assert(false);
-    return null;
+    throw const HijackException();
   }
 
   /// Selects a subprotocol to use for the given connection.
   ///
   /// If no matching protocol can be found, returns `null`.
-  String _chooseProtocol(Request request) {
+  String? _chooseProtocol(Request request) {
     final requestProtocols = request.headers['Sec-WebSocket-Protocol'];
     if (requestProtocols == null) return null;
     if (_protocols == null) return null;
     for (var requestProtocol in requestProtocols.split(',')) {
       requestProtocol = requestProtocol.trim();
-      if (_protocols.contains(requestProtocol)) return requestProtocol;
+      if (_protocols?.contains(requestProtocol) == true) return requestProtocol;
```

Comment:
We can assert this is non-null, it is checked above. You could also assign it to a local variable and check that, making this not necessary.

```suggestion
      if (_protocols!.contains(requestProtocol) == true) return requestProtocol;
```

Reply:
done